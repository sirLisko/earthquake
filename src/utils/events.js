export const filterEvents = (events, filter) =>
  filter.value
    ? events.filter(
        event => event.properties[filter.name].toString() === filter.value,
      )
    : events;

export default { filterEvents };
