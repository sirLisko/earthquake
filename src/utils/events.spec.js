import { filterEvents } from './events';

describe('Events util', () => {
  const events = [
    {
      id: '123',
      properties: { mag: '1', place: 'foo', maagType: 'foo' },
    },
    {
      id: '111',
      properties: { mag: '999', place: 'foo', maagType: 'qwe' },
    },
    {
      id: '321',
      properties: { mag: '01', place: 'foobar', maagType: 'bar' },
    },
  ];

  it('should filter properly', () => {
    const filteredEvents = filterEvents(events, {
      name: 'place',
      value: 'foo',
    });
    expect(filteredEvents).toMatchSnapshot();
    expect(filteredEvents.length).toBe(2);
  });
});
