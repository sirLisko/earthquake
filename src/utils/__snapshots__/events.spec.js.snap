// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Events util should filter properly 1`] = `
Array [
  Object {
    "id": "123",
    "properties": Object {
      "maagType": "foo",
      "mag": "1",
      "place": "foo",
    },
  },
  Object {
    "id": "111",
    "properties": Object {
      "maagType": "qwe",
      "mag": "999",
      "place": "foo",
    },
  },
]
`;
